tutorials/developing_with_ign-cmake.md

```diff
@@ -0,0 +1,197 @@
+\page developingwithcmake Developing with Ignition CMake 
+
+# Developing with Ignition CMake
+
+This tutorial documents various tips and strategies for developing with Ignition CMake.
+
+## Helpful CMake flags
+
+There are several flags that control the results of the CMake tool.
+Some of these flags are built into CMake, where some are Ignition CMake specific.
+
+### Setting the build type
+
+The `CMAKE_BUILD_TYPE` variable controls the type of binary output from the build stage.
+This will have an impact on the flags passed to the compiler and linker.
+
+The available options are:
+
+  * `RelWithDebInfo` (default): Mostly optimized build, but with debug symbols.
+  * `Debug`: Debug build without optimizations, with debug symbols enabled.
+  * `Release`: Fully optimized build, no debug symbols.
+  * `MinSizeRel`: Fully optimized build, minimal binary size 
+  * `Coverage`: Build with additional information required for code coverage computation
+  * `Profile`: Use flags that are helpful with the `gprof` profiling tool
+
+More information about flags applied can be found in [IgnSetCompilerFlags.cmake](https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/cmake/IgnSetCompilerFlags.cmake)
+
+### Creating a compilation database
+
+`CMake` can optionally generate a compilation data base that may be used with a variety of code completion tools.
```

Comment:
```suggestion
`CMake` can optionally generate a compilation database that may be used with a variety of code completion tools.
```

Should we include a link to compilation databases as well? I've always referred to this one, but maybe there's something better out there: https://clang.llvm.org/docs/JSONCompilationDatabase.html